lib/merge.ts

```diff
@@ -159,14 +159,14 @@ export class MergedTypeParameter {
   toTypeParameterDeclaration(): ts.TypeParameterDeclaration {
     let ret = <ts.TypeParameterDeclaration>ts.createNode(ts.SyntaxKind.TypeParameter);
     let nameIdentifier = <ts.Identifier>ts.createNode(ts.SyntaxKind.Identifier);
-    nameIdentifier.text = this.name;
+    nameIdentifier.escapedText = ts.escapeLeadingUnderscores(this.name);
```

Comment:
What does this escapeLeadingUnderscores method do? I would expect we want to be using a Dart specific scheme for escaping leading underscores rather than some typescript specific scheme unless I am misinterpreting where this method is defined.

Reply:
I think this only handles a special case where the identifier begins with two underscores. In typescript.d.ts it says "add an extra underscore to identifiers that start with two underscores to avoid issues with magic names like '__proto__'." I used this because the `text` field on most Nodes changed to `escapedText`, which is of type `__String` instead of string. Since nodes now have their names stored as `__String`s I think calling node.getText() returns unescapeLeadingUnderscores(escapedText), so I didn't want to accidentally miss an underscore.

Reply:
Can you add a test that covers this case?

Reply:
I'm skeptical that is really the obfuscation you want for Dart but I'm not sure exactly what it does.